pulp_rpm/tests/functional/api/test_crud_content_unit.py

```diff
@@ -1,221 +1,143 @@
+    content_unit = rpm_package_api.read(content_unit_original["pulp_href"]).to_dict()
+    for key, val in content_unit_original.items():
+        assert content_unit[key] == val
+
+    # Read a content unit by its pkg_id
+    page = rpm_package_api.list(pkg_id=content_unit_original["pkg_id"])
+    assert len(page.results) == 1
+    for key, val in content_unit_original.items():
+        assert page.results[0].to_dict()[key] == val
+
+    # Attempt to update a content unit using HTTP PATCH
+    attrs = gen_rpm_content_attrs(signed_artifact, RPM_PACKAGE_FILENAME2)
+    with pytest.raises(AttributeError) as exc:
+        rpm_package_api.partial_update(content_unit_original["pulp_href"], attrs)
+    msg = "object has no attribute 'partial_update'"
+    assert msg in str(exc)
+
+    # Attempt to update a content unit using HTTP PUT
+    attrs = gen_rpm_content_attrs(signed_artifact, RPM_PACKAGE_FILENAME2)
+    with pytest.raises(AttributeError) as exc:
+        rpm_package_api.update(content_unit_original["pulp_href"], attrs)
+    msg = "object has no attribute 'update'"
+    assert msg in str(exc)
+
+    # Attempt to delete a content unit using HTTP DELETE
+    with pytest.raises(AttributeError) as exc:
+        rpm_package_api.delete(content_unit_original["pulp_href"])
+    msg = "object has no attribute 'delete'"
+    assert msg in str(exc)
+
+    # Attempt to create duplicate package without specifying a repository
+    attrs = gen_rpm_content_attrs(signed_artifact, RPM_PACKAGE_FILENAME)
+    response = rpm_package_api.create(**attrs)
+    duplicate = rpm_package_api.read(monitor_task(response.task).created_resources[0])
+    assert duplicate.pulp_href == content_unit_original["pulp_href"]
+
+    # Attempt to create duplicate package while specifying a repository
+    repo = rpm_repository_factory()
+    attrs = gen_rpm_content_attrs(signed_artifact, RPM_PACKAGE_FILENAME)
+    attrs["repository"] = repo.pulp_href
+    response = rpm_package_api.create(**attrs)
+    monitored_response = monitor_task(response.task)
+
+    duplicate = rpm_package_api.read(monitored_response.created_resources[1])
+    assert duplicate.pulp_href == content_unit_original["pulp_href"]
+
+    repo = rpm_repository_api.read(repo.pulp_href)
+    assert repo.latest_version_href.endswith("/versions/1/")
+
+    version = rpm_repository_version_api.read(repo.latest_version_href)
+    assert version.content_summary.present["rpm.package"]["count"] == 1
+    assert version.content_summary.added["rpm.package"]["count"] == 1
+
+
+@pytest.mark.parallel
+@pytest.mark.parametrize(
+    "url",
+    [RPM_MODULAR_FIXTURE_URL, RPM_KICKSTART_FIXTURE_URL, RPM_REPO_METADATA_FIXTURE_URL],
+    ids=["MODULAR_FIXTURE_URL", "KICKSTART_FIXTURE_URL", "REPO_METADATA_FIXTURE_URL"],
+)
+def test_remove_content_unit(
+    url, init_and_sync, rpm_repository_version_api, bindings_cfg, http_get
+):
+    """
+    Sync repository and test that content can't be removed directly.
```

Comment:
We should better describe how this is different from the the delete portion of the "crud" test above (line 68), I think this one tests every type of content unit whereas the other only tests packages, is that it?

Reply:
Ah, I have forgotten about this. Will do.